dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/am/KubernetesApplicationManager.java

```diff
@@ -131,6 +132,8 @@ private FilterWatchListDeletable<Pod, PodList, PodResource> getListenPod(Kuberne
     private KubernetesClient getClient(KubernetesApplicationManagerContext kubernetesApplicationManagerContext) {
         K8sTaskExecutionContext k8sTaskExecutionContext =
                 kubernetesApplicationManagerContext.getK8sTaskExecutionContext();
+        k8sTaskExecutionContext
+                .setConfigYaml(JSONUtils.getNodeString(k8sTaskExecutionContext.getConnectionParams(), "kubeConfig"));
```

Comment:
You can not directly change this. For normal task, like spark on k8s, `K8sTaskExecutionContext` is constructed by `TaskExecutionContextFactory`, when they call `getClient`, `connectionParams` is null.

Reply:
Thank you for your suggestion